packages/synapse-react-client/src/components/widgets/query-filter/RangeFacetFilter.tsx

```diff
@@ -60,16 +63,29 @@ export const RangeFacetFilter: React.FunctionComponent<
 
   const rangeType = columnType === 'DOUBLE' ? 'number' : 'date'
 
-  const handleRadioGroupChange = (
-    radioValue: RadioValuesEnum,
-    onChangeCallback: (
-      range: (RadioValuesEnum | number | string | undefined)[],
-    ) => void,
-  ) => {
+  const handleRadioGroupChange = (radioValue: RadioValuesEnum) => {
     setRadioValue(radioValue)
 
-    if (radioValue !== RadioValuesEnum.RANGE) {
-      onChangeCallback([radioValue, radioValue])
+    switch (radioValue) {
+      case RadioValuesEnum.RANGE:
+        // The range facet value will update when the user clicks the apply button
+        break
+      case RadioValuesEnum.NOT_SET:
+        setRangeFacetValue(facetResult, radioValue, radioValue)
+        break
+      case RadioValuesEnum.ANY: {
+        const selectedFacet =
+          getCurrentQueryRequest().query.selectedFacets?.find(
+            selectedFacet =>
+              selectedFacet.columnName === facetResult.columnName &&
+              selectedFacet.jsonPath === facetResult.jsonPath,
+          )
+        if (selectedFacet) {
+          removeSelectedFacet(selectedFacet)
+        }
+
+        break
+      }
```

Comment:
Directly change the query using callbacks provided by `QueryContext` to replace custom callbacks that were previously defined in `FacetFilterControls.tsx` and passed in